aten/src/ATen/native/ResizeCommon.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <ATen/ATen.h>
+#include <ATen/NamedTensorUtils.h>
+
+namespace at { namespace native {
+
+#ifdef BUILD_NAMEDTENSOR
+inline Tensor& resize_named_tensor_(Tensor& self, IntArrayRef size) {
+  TORCH_INTERNAL_ASSERT(self.has_names());
+  TORCH_CHECK(
+      self.sizes() == size,
```

Comment:
would it be possible to re-size from the right if left dimensions are wildcards? e.g. would it make sense for a `tensor([3,4,5], names=(*, *, 'foo')).resize(4,15) -> tensor([4,15], names=(*, 'foo'))` ?

Reply:
In your example, we're "mixing and matching" dimensions with other dimensions. In that case, it's unclear if the last dimension should retain the same name.

For example, someone can do `tensor([5], names=('foo',).resize(5, 1)` -> clearly the first dimension should now be the "foo" dimension and the dimension of size 1 is a brand new dimension.